app/experimenter/experiments/serializers/design.py

```diff
@@ -142,20 +181,12 @@ def validate_preferences(self, data):
         self.validate_value_type_match(data)
         return data
 
-    def is_pref_valid(self, preferences):
-        unique_names = len(
-            set([slugify(pref["pref_name"]) for pref in preferences])
-        ) == len(preferences)
-
-        return unique_names
-
     def validate_value_type_match(self, preferences):
         error_list = []
         for pref in preferences:
-            pref_type = pref.get("pref_type", "")
-            pref_value = pref["pref_value"]
-            field_name = "pref_value"
-            error_list.append(self.validate_pref(pref_type, pref_value, field_name))
+            errors = self.validate_pref_branch(pref["pref_branch"])
```

Comment:
Build the empty errors dict just above this `errors = {}` and then update it with both responses so it doesnt' look like they're order dependent.